r/R/util.R

```diff
@@ -139,3 +139,42 @@ attr(is_writable_table, "fail") <- function(call, env){
   )
 }
 
+#' Take an object of length 1 and repeat it.
+#' 
+#' @param object Object of length 1 to be repeated - vector, `Scalar`, `Array`, or `ChunkedArray`
+#' @param n Number of repetitions
+#' 
+#' @return `Array` of length `n`
+#' 
+#' @keywords internal
+repeat_value_as_array <- function(object, n) {
+  if (inherits(object, "ChunkedArray")) {
+    return(Scalar$create(object$chunks[[1]])$as_array(n))
+  }
+  return(Scalar$create(object)$as_array(n))
+}
+
+#' Recycle scalar values in a list of arrays
+#' 
+#' @param arrays List of arrays
+#' @return List of arrays with any vector/Scalar/Array/ChunkedArray values of length 1 recycled 
+#' @keywords internal
+recycle_scalars <- function(arrays){
+  # Get lengths of items in arrays
+  arr_lens <- map_int(arrays, NROW)
+  
+  if (length(arrays) > 1 && any(arr_lens == 1) && !all(arr_lens==1)) {
+    
+    # Recycling not supported for tibbles and data.frames
+    if(all(map_lgl(arrays, ~inherits(.x, "data.frame")))){
```

Comment:
You added whitespace throughout the package but not here :)

```suggestion
    if (all(map_lgl(arrays, ~inherits(.x, "data.frame")))) {
```